lib/salus/scanners/osv/base.rb

```diff
@@ -0,0 +1,137 @@
+require 'salus/scanners/base'
+require 'json'
+require 'zip'
+require 'uri'
+require 'net/http'
+
+module Salus::Scanners::OSV
+  class Base < Salus::Scanners::Base
+    DATABASE_STRING_MAPPING = {
+      "GHSA" => "Github Advisory Database",
+      "PYSEC" => "Python Packaging Advisory Database",
+      "GO" => "Go Vulnerability Database",
+      "RUSTSEC" => "RustSec Advisory Database",
+      "default" => "Open Source Vulnerabilitiy"
+    }.freeze
+
+    def run
+      raise NoMethodError, 'implement in subclass'
+    end
+
+    def should_run?
+      raise NoMethodError, 'implement in subclass'
+    end
+
+    def name
+      self.class.name.sub('Salus::Scanners::OSV::', '')
+    end
+
+    private
+
+    def osv_vulnerabilities
+      @osv_vulnerabilities ||= fetch_vulnerabilities
+    end
+
+    def osv_urls
+      urls = []
+      urls.append(osv_url_for("Go")) if @repository.go_sum_present? || @repository.go_mod_present?
+      urls.append(osv_url_for("PyPI")) if @repository.requirements_txt_present?
+      urls.append(osv_url_for("Maven")) if @repository.pom_xml_present? ||
+        @repository.build_gradle_present?
+
+      urls
+    end
+
+    def osv_url_for(package)
+      # zip contains individual entries for an ecosystem in OSV format.
+      # Approximate zip sizes
+      # Go: all.zip (478 KB) -> 1.1 MB
+      # PyPI: all.zip (3.4 MB) -> 10.9 MB
+      # Maven: all.zip (1.6 MB) -> 5.1 MB
+      url = "https://osv-vulnerabilities.storage.googleapis.com/"\
+      + package + "/all.zip"
+      URI(url)
+    end
+
+    def fetch_vulnerabilities
+      # Flatten vulnerabilities by package name.
+      all_vulnerabilities = send_request
```

Comment:
but `send_request` could raise an exception. That exception should be checked here.